slither.config.json

```diff
@@ -1,5 +1,4 @@
 {
   "filter_paths": "ERC20.sol|Token.sol|USDC|mocks|Test|RescueNonOptimized|BytesLib",
-  "detectors_to_exclude": "assembly,similar-names,too-many-digits,naming-convention,write-after-write,solc-version",
-  "sarif": "slither.sarif"
+  "detectors_to_exclude": "assembly,similar-names,too-many-digits,naming-convention,write-after-write,solc-version"
```

Comment:
I think [write-after-write](https://github.com/crytic/slither/wiki/Detector-Documentation#write-after-write) may be useful but it has many false positives for the Rescue library. There does not seem to be a way to disable a detector for an entire file. We could also use the [triage mode](https://github.com/crytic/slither/wiki/Usage#triage-mode) to silence the warnings. I will give this a go to see if it works at all.

```
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#185)
	s0 = mulmod(uint256,uint256,uint256)(tmp,tmp,_PRIME) (contracts/libraries/RescueLib.sol#190)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#185)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_0,tmp_scope_0,_PRIME) (contracts/libraries/RescueLib.sol#197)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#185)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_1,tmp_scope_1,_PRIME) (contracts/libraries/RescueLib.sol#204)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#185)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_2,tmp_scope_2,_PRIME) (contracts/libraries/RescueLib.sol#211)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#227)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_3,tmp_scope_3,_PRIME) (contracts/libraries/RescueLib.sol#232)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#227)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_4,tmp_scope_4,_PRIME) (contracts/libraries/RescueLib.sol#239)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#227)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_5,tmp_scope_5,_PRIME) (contracts/libraries/RescueLib.sol#246)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#227)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_6,tmp_scope_6,_PRIME) (contracts/libraries/RescueLib.sol#253)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#269)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_7,tmp_scope_7,_PRIME) (contracts/libraries/RescueLib.sol#274)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#269)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_8,tmp_scope_8,_PRIME) (contracts/libraries/RescueLib.sol#281)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#269)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_9,tmp_scope_9,_PRIME) (contracts/libraries/RescueLib.sol#288)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#269)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_10,tmp_scope_10,_PRIME) (contracts/libraries/RescueLib.sol#295)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#311)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_11,tmp_scope_11,_PRIME) (contracts/libraries/RescueLib.sol#316)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#311)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_12,tmp_scope_12,_PRIME) (contracts/libraries/RescueLib.sol#323)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#311)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_13,tmp_scope_13,_PRIME) (contracts/libraries/RescueLib.sol#330)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#311)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_14,tmp_scope_14,_PRIME) (contracts/libraries/RescueLib.sol#337)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#353)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_15,tmp_scope_15,_PRIME) (contracts/libraries/RescueLib.sol#358)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#353)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_16,tmp_scope_16,_PRIME) (contracts/libraries/RescueLib.sol#365)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#353)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_17,tmp_scope_17,_PRIME) (contracts/libraries/RescueLib.sol#372)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#353)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_18,tmp_scope_18,_PRIME) (contracts/libraries/RescueLib.sol#379)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#395)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_19,tmp_scope_19,_PRIME) (contracts/libraries/RescueLib.sol#400)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#395)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_20,tmp_scope_20,_PRIME) (contracts/libraries/RescueLib.sol#407)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#395)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_21,tmp_scope_21,_PRIME) (contracts/libraries/RescueLib.sol#414)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#395)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_22,tmp_scope_22,_PRIME) (contracts/libraries/RescueLib.sol#421)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#437)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_23,tmp_scope_23,_PRIME) (contracts/libraries/RescueLib.sol#442)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#437)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_24,tmp_scope_24,_PRIME) (contracts/libraries/RescueLib.sol#449)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#437)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_25,tmp_scope_25,_PRIME) (contracts/libraries/RescueLib.sol#456)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#437)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_26,tmp_scope_26,_PRIME) (contracts/libraries/RescueLib.sol#463)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#479)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_27,tmp_scope_27,_PRIME) (contracts/libraries/RescueLib.sol#484)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#479)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_28,tmp_scope_28,_PRIME) (contracts/libraries/RescueLib.sol#491)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#479)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_29,tmp_scope_29,_PRIME) (contracts/libraries/RescueLib.sol#498)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#479)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_30,tmp_scope_30,_PRIME) (contracts/libraries/RescueLib.sol#505)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#521)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_31,tmp_scope_31,_PRIME) (contracts/libraries/RescueLib.sol#526)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#521)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_32,tmp_scope_32,_PRIME) (contracts/libraries/RescueLib.sol#533)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#521)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_33,tmp_scope_33,_PRIME) (contracts/libraries/RescueLib.sol#540)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#521)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_34,tmp_scope_34,_PRIME) (contracts/libraries/RescueLib.sol#547)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#563)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_35,tmp_scope_35,_PRIME) (contracts/libraries/RescueLib.sol#568)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#563)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_36,tmp_scope_36,_PRIME) (contracts/libraries/RescueLib.sol#575)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#563)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_37,tmp_scope_37,_PRIME) (contracts/libraries/RescueLib.sol#582)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#563)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_38,tmp_scope_38,_PRIME) (contracts/libraries/RescueLib.sol#589)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#605)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_39,tmp_scope_39,_PRIME) (contracts/libraries/RescueLib.sol#610)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#605)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_40,tmp_scope_40,_PRIME) (contracts/libraries/RescueLib.sol#617)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#605)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_41,tmp_scope_41,_PRIME) (contracts/libraries/RescueLib.sol#624)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#605)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_42,tmp_scope_42,_PRIME) (contracts/libraries/RescueLib.sol#631)
RescueLib.perm(uint256,uint256,uint256,uint256).s0 (contracts/libraries/RescueLib.sol#157) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#647)
	s0 = mulmod(uint256,uint256,uint256)(tmp_scope_43,tmp_scope_43,_PRIME) (contracts/libraries/RescueLib.sol#652)
RescueLib.perm(uint256,uint256,uint256,uint256).s1 (contracts/libraries/RescueLib.sol#158) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#647)
	s1 = mulmod(uint256,uint256,uint256)(tmp_scope_44,tmp_scope_44,_PRIME) (contracts/libraries/RescueLib.sol#659)
RescueLib.perm(uint256,uint256,uint256,uint256).s2 (contracts/libraries/RescueLib.sol#159) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#647)
	s2 = mulmod(uint256,uint256,uint256)(tmp_scope_45,tmp_scope_45,_PRIME) (contracts/libraries/RescueLib.sol#666)
RescueLib.perm(uint256,uint256,uint256,uint256).s3 (contracts/libraries/RescueLib.sol#160) is written in both
	(s0,s1,s2,s3) = _linearOp(s0,s1,s2,s3) (contracts/libraries/RescueLib.sol#647)
	s3 = mulmod(uint256,uint256,uint256)(tmp_scope_46,tmp_scope_46,_PRIME) (contracts/libraries/RescueLib.sol#673)
PlonkVerifier._batchVerifyOpeningProofs(PlonkVerifier.PcsInfo[]).tmp (contracts/verifier/PlonkVerifier.sol#471) is written in both
	tmp = mulmod(uint256,uint256,uint256)(rBase,evalPoint,p) (contracts/verifier/PlonkVerifier.sol#523)
	tmp = mulmod(uint256,uint256,uint256)(rBase,mulmod(uint256,uint256,uint256)(u,nextEvalPoint,p),p) (contracts/verifier/PlonkVerifier.sol#537)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#write-after-write
```

Reply:
Enabled `write-to-write` in https://github.com/EspressoSystems/cape/pull/1109/commits/53293c958b4631e283a7ad7127952b6d6ebb3041 .